### DISCUSSION TITLE
What happens if git registry goes down?

### DISCUSSION THREAD

**Discussion by mihaigalos** [argoproj/argo-cd#21072] [Q&A] [answered]
ArgoCD keeps state of the cluster in sync with the remote git registry (i.e. GitHub, GitHub Enterprise, GitLab).

What would happen in the hypothetical situation that for example GitHub Enterprise goes down?
ArgoCD would not reach the remote to fetch latest HEAD. Does it still enforce the last known state or is does the failure create a halt/stop and the state does not get enforced?

**Comment by rumstead** [comment]
> ArgoCD keeps state of the cluster in sync with the remote git registry (i.e. GitHub, GitHub Enterprise, GitLab).
> 
> What would happen in the hypothetical situation that for example GitHub Enterprise goes down? ArgoCD would not reach the remote to fetch latest HEAD. Does it still enforce the last known state or is does the failure create a halt/stop and the state does not get enforced?

Applications will go into the unknown state and it won't be able to sync.

**Reply by mihaigalos** [reply]
so does it enforce last known state or not?

**Reply by rumstead** [reply, marked as answer]
no, it wouldn't be "enforcing" anything because the application would be in an unknown state and unable to reconcile from git.
